### PR TITLE
FEATURE: Show unpublished changes

### PR DESCRIPTION
**Pull request by mstruebing**
needs https://github.com/sitegeist/Sitegeist.TrafficLights

### CODE REVIEW

File: packages/neos-ui-redux-store/src/CR/Nodes/selectors.js [hunk 1]
```diff
@@ -327,3 +327,37 @@ export const focusedNodeParentLineSelector = createSelector(
...
+        (state, contextPath) => {
+            return contextPath;
+        }
+    ],
+    (nodes, contextPath) => {
+        if (nodes) {
+            return nodes.reduce((workspaces, node) => {
+                if (nodes.filter(i => $get('hasForeignChanges', i) && $get('contextPath', i) === contextPath).count() > 0) {
+                    if ($get('hasForeignChanges', node)) {
+                        workspaces.push($get('foreignWorkspacesWithChanges', node)._tail.array);
```
Comment (mstruebing) [inline packages/neos-ui-redux-store/src/CR/Nodes/selectors.js:353]: wtf?

File: packages/neos-ui-redux-store/src/CR/Nodes/selectors.js [hunk 2]
```diff
@@ -327,3 +327,37 @@ export const focusedNodeParentLineSelector = createSelector(
...
+    [
+        $get('cr.nodes.byContextPath'),
+        (state, contextPath) => {
+            return contextPath;
+        }
+    ],
+    (nodes, contextPath) => {
+        if (nodes) {
+            return nodes.reduce((workspaces, node) => {
+                if (nodes.filter(i => $get('hasForeignChanges', i) && $get('contextPath', i) === contextPath).count() > 0) {
```
Comment (mstruebing) [inline packages/neos-ui-redux-store/src/CR/Nodes/selectors.js:351]: this is somehow the makeHasForeignChangesSelector and should be reused

**Comment by dimaip** [comment]
Could you please briefly describe what this PR does?

**Comment by mstruebing** [comment]
This is only a PR to my fork to summarize what I've done in order to extract that into a plugin.
Basically it shows in the page tree if a document node has unpublished changes from a foreign workspace, on hover you see which user changed something.

We want this for our customer so they don't override each others changes by visualizing that something changed from another editor.

**Comment by mstruebing** [comment]
screenshot_2018-04-26_11-04-13

On hover of the blue marked node you see the users which has unpublished changes to ping them :)

**Comment by dimaip** [comment]
Ah, nice! Sorry, didn't realize it was your local PR :see_no_evil:

**Comment by mstruebing** [comment]
If we polish this thing it is maybe a good feature to have in the core :)

**Pull request closed without merging**